services/search/search_engine.py

```diff
@@ -0,0 +1,99 @@
+import requests # pylint: disable=import-error
+import constants
+import testing.test_constants
+import utils
+
+class SearchEngine:
+    """
+    This class implements the functions to score every article
+    by the amount of incidences between keywords and synonyms to articles
+
+    Attributes:
+        keywords: A list of keywords
+        synonyms: A list of synonyms for the keywords
+        articles: A list of articles
+        keywords_weight: The weight that a keyword has on the scorig algorithm
+            Optional: default argument as 1
+        synonyms_weight: The weight that a synonym has on the scoring algorithm
+            Optional: default argument as 1
+    """
+    def __init__(self, keywords_weight=1, synonyms_weight=1):
+        """
+        Inits the search engine by loading all articles
+        """
+        self.keywords_weight = keywords_weight
+        self.synonyms_weight = synonyms_weight
+
+    def query(self, query):
+        """
+        Extracts keywords, from keywords endpoint,
+        and synonyms, from local function,
+        to call search query and calculate the score for articles
+        Args:
+            query: query string
+        Returns:
+            A map of the score of every article
+        """
+        query_text = {'text': query}
+        response = requests.post(constants.KEYWORDS_ENDPOINT, json=query_text)
+        response = response.json()
+
+        if 'error' in response:
+            return response
+
+        lan = response['lan']
+
+        if lan not in constants.SUPPORTED_LANGUAGES:
```

Comment:
Please re-raise error here and defer error-formatting to calling function. That way this class can also be used by other python libraries, and not just HTTP servers that need to return JSON.